coordinator/store/stdstore/stdstore.go

```diff
@@ -124,10 +159,25 @@ func (s *StdStore) SetRecoveryData(recoveryData []byte) {
 
 // SetEncryptionKey sets the encryption key for sealing and unsealing.
 func (s *StdStore) SetEncryptionKey(encryptionKey []byte) error {
-	return s.sealer.SetEncryptionKey(encryptionKey)
+	// If there already is an existing key file stored on disk, save it
+	s.backupEncryptionKey()
+
+	encryptedKey, err := s.sealer.SetEncryptionKey(encryptionKey)
+	if err != nil {
+		return fmt.Errorf("encrypting data key: %w", err)
+	}
+
+	// Write the sealed encryption key to disk
+	if err = s.fs.WriteFile(filepath.Join(s.sealDir, SealedKeyFname), encryptedKey, 0o600); err != nil {
+		// Reset encryption key if writing to disk fails
+		_, _ = s.sealer.SetEncryptionKey(encryptionKey)
```

Comment:
This sets the new key again, i.e., it's a no-op.
I think we should factor out the respective code into a new func SealEncryptionKey and let SetEncryptionKey only do what it says.
And then do:
SealEncryptionKey()
WriteFile()
SetEncryptionKey()